File: src/Projects/Android/012-ImageProcessingServerNetLib/ConsoleClient/src/main/java/org/csystem/app/imageprocessing/client/runner/ClientRunner.java
package org.csystem.app.imageprocessing.client.runner;

import org.csystem.app.imageprocessing.client.Client;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ClientRunner implements ApplicationRunner {
    private final Client m_client;

    public ClientRunner(Client client)
    {
        m_client = client;
    }

    @Override
    public void run(ApplicationArguments args) throws Exception
    {
        new Thread(m_client::run).start();
    }
}

File: src/Projects/Android/013-RandomTextGeneratorServer/Client/src/main/kotlin/org/csystem/app/random/text/client/runner/ClientRunner.kt
package org.csystem.app.random.text.client.runner

import org.csystem.app.random.text.client.Client
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import java.util.concurrent.ExecutorService

@Component
class ClientRunner(private val mThreadPool: ExecutorService, private val mClient: Client) : CommandLineRunner {
    override fun run(vararg args: String?) {
        mThreadPool.execute { mClient.start() }
        mThreadPool.shutdown()
    }
}